gubernator/kubelet_parser.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import datetime
+import os
+import re
+
+import jinja2
+
+
+def hilight(line, error_re):
+    line = error_re.sub(r'<span class="keyword">\1</span>', line)
+    return '<span class="hilight">%s</span>' % line
+
+
+def digest(data, pod, skip_fmt=lambda l: '... skipping %d lines ...' % l):
```

Comment:
It's fine to put the kubelet log matcher in a separate file, just avoid duplicating the other fiddly annotation context logic.